docs/source/contributing.rst

```diff
@@ -4,88 +4,139 @@ Contributing
+   You can use SSH if you have SSH keys setup.
 
    .. WARNING::
 
-      Do not ``git clone`` the original ManimCommunity repository.  You must
-      clone your own fork.  After this step, there are three different
-      repositories to keep track of: the original ManimCommunity repo, your own
-      fork of it, and your local repository.
+      Do not clone the ManimCommunity repository. You must clone your own
+      fork.
+
+#.  Change the directory to enter the project folder:
+
+    .. code-block:: shell
 
-#. Make ``git`` aware of the ManimCommunity repo.
+       cd manim
+
+#. Add the upstream repository, ManimCommunity:
 
    .. code-block:: shell
 
       git remote add upstream https://github.com/ManimCommunity/manim.git
-      git fetch upstream
+ 
+#. Now, ``git remote -v`` should show two remote repositores named:
+   
+   - ``origin``, your forked repository
+   - ``upstream`` the ManimCommunity repository
+
+#. Install Manim:
+ 
+   - See our :doc:`instructions for developers<installation/for_dev>` for
+     details and continue here afterwards
+
+#. Install Pre-Commit (recommended):
 
-   After these commands, your local repository can keep track of your fork
-   (referred to as 'origin') as well as the main ManimCommunity repository
-   (referred to as 'upstream').
+   .. code-block:: shell
 
-#. *Install manim.*
-   See the :doc:`installation instructions for developers <installation/for_dev>` for
-   details.
+      poetry run pre-commit install
+   
+   This is optional and will ensure during development that each of your
+   commit is properly formatted against our formatter, ``black``.
 
 You are now ready to work on manim!
 
+Develop your contribution
+--------------------------
+
+#. Checkout your local repository's master branch and pull the latest
+   changes from ManimCommunity, `upstream`, into your local repository:
+
+   .. code-block:: shell
 
-Changing manim's source code
-============================
+      git checkout master 
+      git pull --rebase upstream master
 
-#. *Choose the branch for your changes.*
-   To work on the ManimCommunity master branch, you can change to it with:
+#. Create a branch for the changes you want to work on rather than working
+   off of master:
 
    .. code-block:: shell
 
-      git checkout -b master upstream/master
+      git checkout -b <new branch name> upstream/master
+
+   This ensures you can easily update your local repository's master with the
+   first step and switch branches to work on multiple features.
+
+#. Add new or update existing test.
+
+   Depending on your changes, you may need to update or add new tests. For new
+   features, it is required that you include tests with your PR. Details of
+   our testing system are explained in the `corresponding Wiki Entry <https://github.com/ManimCommunity/manim/wiki/Testing>`_.
 
-   If you are starting a new branch, execute
+#. Write some awesome code!
```

Comment:
I feel that this should come before the "Add new or update existing tests"-part.

Reply:
I was mostly thinking of test first programming/test driven development, but since testing in Manim is difficult to do (control data for graphical/video tests isn't usually possible to generate) before the feature is actually implemented, I'll switch the ordering